Foundation/NSURLSession/HTTPBodySource.swift

```diff
@@ -0,0 +1,261 @@
+// Foundation/NSURLSession/HTTPBodySource.swift - NSURLSession & libcurl
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// These are libcurl helpers for the NSURLSession API code.
+/// - SeeAlso: https://curl.haxx.se/libcurl/c/
+/// - SeeAlso: NSURLSession.swift
+///
+// -----------------------------------------------------------------------------
+
+import CoreFoundation
+import Dispatch
+
+
+
+internal extension NSData {
+    /// Turn `dispatch_data_t` into `NSData`
+    convenience init(dispatchData: dispatch_data_t) {
+        //TODO: Should do this through an NSData subclass to avoid copying
```

Comment:
In the Swift 2.x API for Dispatch we have a `dispatch_data_t` class that wraps the C implementation.